### PR TITLE
keycloak improvements

### PR DESCRIPTION
**Pull request by depahelix2021**
keycloak is super super slow on Apple Silicon, to the point where it gums up the whole startup process for Chorus.  Setting the right platform and building the keycloak image does the trick.  This is the recommended method anyway, so let's do it all the time, but make sure we pass in the correct platform string to docker-compose.yml via environment variables, then set an arg for docker build for the admin password too, and then pass that down through the use of ARG and ENV in the Dockerfile.  This way the admin password for keycloak can be changed at the top level without much fuss.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -296,19 +296,17 @@ services:
       - COLLECTOR_OTLP_ENABLED=true
 
   keycloak:
-    image: quay.io/keycloak/keycloak:20.0.3
```
Comment (risdenk) [inline docker-compose.yml:299]: Based on https://quay.io/repository/keycloak/keycloak?tab=tags&tag=20.0.3 if you look at the child manifests - there is a Docker image pushed with amd64 / arm64. Docker should pull the correct image. Building an image doesn't change that logic (you use the same Dockerfile FROM as image here)

File: quickstart.sh [hunk 2]
```diff
@@ -28,6 +31,10 @@ do
 			offline_lab=true
       log_major "Running Chorus with offline lab environment enabled"
 			;;
+    --apple-silicon | -apple)
+      apple_silicon=true
+      log_major "Configuring Chorus for Apple Silicon"
+      ;;
```
Comment (risdenk) [inline quickstart.sh:37]: I know mysql was fixed without needing this flag. Docker should just work to determine the right image to pull.

**Review by risdenk — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (risdenk) [reply docker-compose.yml:299]: I checked that pulling with `docker compose pull` and then checking `docker inspect quay.io/keycloak/keycloak:20.0.3` that `keycloak` architecture is `"Architecture": "arm64",` on m1 mac.

File: quickstart.sh (hunk 2 above)
Comment (epugh) [reply quickstart.sh:37]: The fewer flags to the `quickstart.sh`, the better!   It's meant for a newbie to quickly use, so we don't want too many options...  every option just makes it more confusing....    Maybe someday we'll have a `powerstart.sh`  script that does all the things we would want if Chorus was actually a project that could be deployed in a production like environment versus a learning tool!

File: docker-compose.yml [hunk 3]
```diff
@@ -296,19 +296,17 @@ services:
...
+    platform: ${KEYCLOAK_DOCKER_PLATFORM}
+    image: chorus-keycloak
     container_name: keycloak
-    hostname: keycloak
+    build:
+      context: ./keycloak
+      dockerfile: Dockerfile
+      args:
+        - keycloak_admin_password=${KEYCLOAK_ADMIN_PASSWORD}
```
Comment (epugh) [inline docker-compose.yml:306]: I am okay with hardcoding these thigns.  IN Chorus, the admin password will always be "password"...  and the admin username will laways either be "admin" or "admin@choruselectronics.com" ;-)    Fewer parameters and fewer configuration will help folks easier udnerstand what is happening...

**Comment by epugh** [comment]
@depahelix2021 @risdenk would love another set of eyes...   i think i threaded the needle of simplicity, http access to keycloak, and FAST startup wiht our preconfigured realms.

**Review by risdenk — APPROVED** [review]
I'm not 100% sure we need to build the keycloak dockerfile, but looks good

**Comment by depahelix2021** [comment]
Looks good, ship it!

**Pull request merged**